upup/pkg/fi/cloudup/scalewaytasks/instance.go

```diff
@@ -287,3 +290,55 @@ func (_ *Instance) RenderScw(c *fi.CloudupContext, actual, expected, changes *In
 
 	return nil
 }
+
+type terraformInstanceIP struct{}
+
+type terraformInstance struct {
+	Name     *string                             `cty:"name"`
+	IPID     *terraformWriter.Literal            `cty:"ip_id"`
+	Type     *string                             `cty:"type"`
+	Tags     []string                            `cty:"tags"`
+	Image    *string                             `cty:"image"`
+	UserData map[string]*terraformWriter.Literal `cty:"user_data"`
+}
+
+func (_ *Instance) RenderTerraform(t *terraform.TerraformTarget, actual, expected, changes *Instance) error {
+	tfName := strings.Replace(fi.ValueOf(expected.Name), ".", "-", -1)
+
+	tfInstanceIP := terraformInstanceIP{}
+	err := t.RenderResource("scaleway_instance_ip", tfName, tfInstanceIP)
+	if err != nil {
+		return err
+	}
+
+	tfInstance := terraformInstance{
+		Name:  expected.Name,
+		IPID:  tfInstanceIP.TerraformLink(tfName),
```

Comment:
I think it is the only place where it's used, so should be safe to inline.
```suggestion
		IPID:  terraformWriter.LiteralProperty("scaleway_instance_ip", tfName, "id"),
```